diff -Naur TFT_eSPI/Processors/TFT_eSPI_ESP32.h TFT_eSPI/Processors/TFT_eSPI_ESP32.h
--- TFT_eSPI/Processors/TFT_eSPI_ESP32.h	2022-11-11 14:08:30.000000000 +0100
+++ TFT_eSPI/Processors/TFT_eSPI_ESP32.h	2022-11-13 20:44:22.000000000 +0100
@@ -206,8 +206,10 @@
 ////////////////////////////////////////////////////////////////////////////////////////
 #ifndef TFT_CS
   #define TFT_CS -1  // Keep DMA code happy
-  #define CS_L       // No macro allocated so it generates no code
-  #define CS_H       // No macro allocated so it generates no code
+//  #define CS_L       // No macro allocated so it generates no code
+//  #define CS_H       // No macro allocated so it generates no code
+  #define CS_L GPIO.out_w1tc = (TFT_CS_MASK); GPIO.out_w1tc = (TFT_CS_MASK)
+  #define CS_H GPIO.out_w1ts = (TFT_CS_MASK)//;GPIO.out_w1ts = (TFT_CS_MASK)
 #else
   #if defined (TFT_PARALLEL_8_BIT)
     #if TFT_CS >= 32
diff -Naur TFT_eSPI/TFT_eSPI.h TFT_eSPI/TFT_eSPI.h
--- TFT_eSPI/TFT_eSPI.h	2022-11-11 14:08:30.000000000 +0100
+++ TFT_eSPI/TFT_eSPI.h	2022-11-13 20:44:33.000000000 +0100
@@ -415,6 +415,8 @@
  //--------------------------------------- public ------------------------------------//
  public:
 
+  uint32_t TFT_CS_MASK;
+
   TFT_eSPI(int16_t _W = TFT_WIDTH, int16_t _H = TFT_HEIGHT);
 
   // init() and begin() are equivalent, begin() included for backwards compatibility
diff -Naur TFT_eSPI/User_Setups/Setup70_ESP32_S2_ILI9341.h TFT_eSPI/User_Setups/Setup70_ESP32_S2_ILI9341.h
--- TFT_eSPI/User_Setups/Setup70_ESP32_S2_ILI9341.h	2022-11-11 14:08:27.000000000 +0100
+++ TFT_eSPI/User_Setups/Setup70_ESP32_S2_ILI9341.h	2022-11-13 20:45:07.000000000 +0100
@@ -4,30 +4,40 @@
 // See SetupX_Template.h for all options available
 #define ILI9341_DRIVER
 
-                    // Typical board default pins
-#define TFT_CS   10 //     10 or 34
+// Define to disable all #warnings in library (can be put in User_Setup_Select.h)
+#define DISABLE_ALL_LIBRARY_WARNINGS
 
-#define TFT_MOSI 11 //     11 or 35
-#define TFT_SCLK 12 //     12 or 36
-#define TFT_MISO 13 //     13 or 37
+#define TFT_WIDTH  240
+#define TFT_HEIGHT 320
 
-#define TFT_DC   14
-#define TFT_RST  15
+#define MULTI_TFT_SUPPORT
+
+// Typical board default pins
+//#define TFT_CS  26 //     10 or 34
+//#define TFT_CS   5 //     10 or 34
+
+#define TFT_MOSI 23 //     11 or 35
+#define TFT_SCLK 18 //     12 or 36
+#define TFT_MISO 19 //     13 or 37
+
+#define TFT_DC   2
+#define TFT_RST  -1
+//#define TFT_RST  11
 
 //#define TOUCH_CS 16 // Optional for touch screen
 
-#define LOAD_GLCD
-#define LOAD_FONT2
-#define LOAD_FONT4
-#define LOAD_FONT6
-#define LOAD_FONT7
-#define LOAD_FONT8
+//#define LOAD_GLCD
+//#define LOAD_FONT2
+//#define LOAD_FONT4
+//#define LOAD_FONT6
+//#define LOAD_FONT7
+//#define LOAD_FONT8
 #define LOAD_GFXFF
 
 #define SMOOTH_FONT
 
 // FSPI port will be used unless the following is defined
-#define USE_HSPI_PORT
+//#define USE_HSPI_PORT
 
 //#define SPI_FREQUENCY  27000000
 #define SPI_FREQUENCY  40000000   // Maximum for ILI9341
diff -Naur TFT_eSPI/User_Setup_Select.h TFT_eSPI/User_Setup_Select.h
--- TFT_eSPI/User_Setup_Select.h	2022-11-11 14:08:30.000000000 +0100
+++ TFT_eSPI/User_Setup_Select.h	2022-11-13 20:45:42.000000000 +0100
@@ -27,7 +27,7 @@
 
 // Only ONE line below should be uncommented to define your setup.  Add extra lines and files as needed.
 
-#include <User_Setup.h>           // Default setup is root library folder
+//#include <User_Setup.h>           // Default setup is root library folder
 
 //#include <User_Setups/Setup1_ILI9341.h>  // Setup file for ESP8266 configured for my ILI9341
 //#include <User_Setups/Setup2_ST7735.h>   // Setup file for ESP8266 configured for my ST7735
@@ -93,7 +93,7 @@
 //#include <User_Setups/Setup61_RP2040_ILI9341_PIO_SPI.h>    // Setup file for RP2040 with PIO SPI ILI9341
 //#include <User_Setups/Setup62_RP2040_Nano_Connect_ILI9341.h>    // Setup file for RP2040 with SPI ILI9341
 
-//#include <User_Setups/Setup70_ESP32_S2_ILI9341.h>     // Setup file for ESP32 S2 with SPI ILI9341
+#include <User_Setups/Setup70_ESP32_S2_ILI9341.h>     // Setup file for ESP32 S2 with SPI ILI9341
 //#include <User_Setups/Setup70b_ESP32_S3_ILI9341.h>    // Setup file for ESP32 S3 with SPI ILI9341
 //#include <User_Setups/Setup70c_ESP32_C3_ILI9341.h>    // Setup file for ESP32 C3 with SPI ILI9341
 //#include <User_Setups/Setup70d_ILI9488_S3_Parallel.h> // Setup file for ESP32 S3 with SPI ILI9488
